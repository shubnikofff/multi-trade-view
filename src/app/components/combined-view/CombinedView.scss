@import "../../variables";

.combined-view {
  height: 100%;
  display: grid;
  grid-template-columns: 1fr 2fr 1fr;

  & > div:first-child {
    border-right: 1px solid $color-border;
  }

  & > div:nth-child(3) {
    border-left: 1px solid $color-border;
  }
}
