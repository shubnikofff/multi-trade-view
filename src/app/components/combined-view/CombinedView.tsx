import React from 'react';

import { useParams } from 'react-router-dom';

import Chat from '../chat';
import Dashboard from '../dashboard';
import TradeList from '../trade-list';

import './CombinedView.scss';

function CombinedView() {
    const { tradeId } = useParams();

    return (
        <div className="combined-view">
            <TradeList />
            {tradeId && <Chat />}
            {tradeId && <Dashboard />}
        </div>
    );
}

export default CombinedView;
