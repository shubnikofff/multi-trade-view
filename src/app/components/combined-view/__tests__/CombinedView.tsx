import React from 'react';

import { mount } from 'enzyme';
import { store } from '../../../moockStore';

import CombinedView from '../CombinedView';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

describe('Component CombinedView', () => {
    const wrapper = mount(
        <Provider store={store}>
            <BrowserRouter>
                <CombinedView />
            </BrowserRouter>
        </Provider>
    );

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
