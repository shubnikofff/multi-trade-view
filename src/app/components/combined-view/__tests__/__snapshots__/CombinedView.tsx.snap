// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CombinedView should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <CombinedView>
        <div
          className="combined-view"
        >
          <TradeList>
            <div
              className="trade-list"
            >
              <Link
                key="1"
                to="/chat/1"
              >
                <LinkAnchor
                  href="/chat/1"
                  navigate={[Function]}
                >
                  <a
                    href="/chat/1"
                    onClick={[Function]}
                  >
                    <div
                      className="trade-list__item"
                      key="1"
                    >
                      <div
                        className="trade-list__item__left-side"
                      >
                        <div
                          className="text-secondary"
                        >
                          <span
                            className="dot dot_gray"
                          />
                        </div>
                        <div>
                          <div
                            className="trade-list__item__left-side__buyer"
                          >
                            <b>
                              username is buying
                            </b>
                          </div>
                          <div>
                            <b>
                              Amazon Gift Card
                            </b>
                          </div>
                          <div
                            className="trade-list__item__left-side__amount text-secondary"
                          >
                            100 USD (0.01000000 BTC)
                          </div>
                        </div>
                      </div>
                      <div
                        className="trade-list__item__right-side"
                      >
                        <div>
                          <Avatar
                            url="url"
                          >
                            <img
                              alt="Avatar"
                              className="avatar"
                              src="url"
                            />
                          </Avatar>
                        </div>
                        <div
                          className="trade-list__item__status trade-list__item__status_gray"
                        >
                          <b>
                            Not paid
                          </b>
                        </div>
                      </div>
                    </div>
                  </a>
                </LinkAnchor>
              </Link>
              <Link
                key="2"
                to="/chat/2"
              >
                <LinkAnchor
                  href="/chat/2"
                  navigate={[Function]}
                >
                  <a
                    href="/chat/2"
                    onClick={[Function]}
                  >
                    <div
                      className="trade-list__item"
                      key="2"
                    >
                      <div
                        className="trade-list__item__left-side"
                      >
                        <div
                          className="text-secondary"
                        >
                          <span
                            className="dot dot_gray"
                          />
                        </div>
                        <div>
                          <div
                            className="trade-list__item__left-side__buyer"
                          >
                            <b>
                              username is buying
                            </b>
                          </div>
                          <div>
                            <b>
                              iTunes Gift Card
                            </b>
                          </div>
                          <div
                            className="trade-list__item__left-side__amount text-secondary"
                          >
                            150 USD (0.01500000 BTC)
                          </div>
                        </div>
                      </div>
                      <div
                        className="trade-list__item__right-side"
                      >
                        <div>
                          <Avatar
                            url="url"
                          >
                            <img
                              alt="Avatar"
                              className="avatar"
                              src="url"
                            />
                          </Avatar>
                        </div>
                        <div
                          className="trade-list__item__status trade-list__item__status_green"
                        >
                          <b>
                            Paid
                          </b>
                        </div>
                      </div>
                    </div>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </TradeList>
        </div>
      </CombinedView>
    </Router>
  </BrowserRouter>
</Provider>
`;
